tests/api_tests/comments/test_views.py

```diff
@@ -0,0 +1,889 @@
+from nose.tools import *  # flake8: noqa
+
+from api.base.settings.defaults import API_BASE
+
+from tests.base import ApiTestCase
+from tests.factories import ProjectFactory, AuthUserFactory, CommentFactory, RegistrationFactory
+
+
+class TestCommentDetailView(ApiTestCase):
+    def setUp(self):
+        super(TestCommentDetailView, self).setUp()
+        self.user = AuthUserFactory()
+        self.contributor = AuthUserFactory()
+        self.non_contributor = AuthUserFactory()
+
+        self.private_project = ProjectFactory(is_public=False, creator=self.user)
```

Comment:
Most of these tests use only one or two of these values. Since setup runs before every test, this means a lot of extra work is being done to create and destroy projects that are not being used. You can move some of that logic inside helper methods (anything not prefixed with `test_` doesn't get run as a test) to reduce redundant code withgout all the extra DB work on every test.